operator/pkg/common/extensions/mapkubeapis.go

```diff
@@ -0,0 +1,78 @@
+package extensions
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	mapkubeapisCommon "github.com/helm/helm-mapkubeapis/pkg/common"
+	mapkubeapisV3 "github.com/helm/helm-mapkubeapis/pkg/v3"
+	"github.com/operator-framework/helm-operator-plugins/pkg/extensions"
+	"github.com/pkg/errors"
+	platform "github.com/stackrox/rox/operator/apis/platform/v1alpha1"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// MapKubeAPIsExtensionConfig extension configuration
+type MapKubeAPIsExtensionConfig struct {
+	KubeConfig mapkubeapisCommon.KubeConfig
+	MapFile    string
+}
+
+// MapKubeAPIsExtension checks the latest release version for any deprecated or removed APIs and performs
+// // the cleanup using helm mapkubeapis extension
+func MapKubeAPIsExtension() extensions.ReconcileExtension {
+	configDir := os.Getenv("OPERATOR_CONFIG_DIR")
+	if configDir == "" {
+		configDir = "config"
+	}
+	config := MapKubeAPIsExtensionConfig{
+		MapFile: filepath.Join(configDir, "mapkubeapis", "Map.yaml"),
```

Comment:
Would it be better if we inject the path to the Map.yaml instead of the config dir?
eg.
```
MAPKUBEAPIS_PATH=/bla/Map.yaml operator

...
MapFile: os.GetEnv("MAPKUBEAPIS_PATH")
```